app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -165,7 +174,18 @@ def input_vars_type_constraints
     payload = JSON.parse(template.payload)
     (payload['input_vars'] || []).index_by { |v| v['name'] }
   rescue => error
-    _log.error("Failure in parsing payload for template/#{template.id}, caused by #{error.message}")
+    $embedded_terraform_log.error("Failure in parsing payload for template/#{template.id}, caused by #{error.message}")
     {}
   end
+
+  def terraform_runner_action_type(resource_action)
+    case resource_action
+    when ResourceAction::RECONFIGURE
+      Terraform::Runner::ActionType::UPDATE
+    when ResourceAction::RETIREMENT
+      Terraform::Runner::ActionType::DELETE
+    else
+      Terraform::Runner::ActionType::CREATE
```

Comment:
Really minor, but if I pass in garbage it would translate to CREATE.  If you're trying for a 1:1 mapping I like an explicit last `when` and the `else` fallthrough is a `raise ArgumentError, "Invalid resource_action type #{resource_action}"`